src/Microsoft.OData.Client/Metadata/ClientTypeUtil.cs

```diff
@@ -687,6 +698,72 @@ internal static string GetEnumValuesString(string enumString, Type enumType)
             return string.Join(",", memberValues);
         }
 
+        /// <summary>
+        /// Returns true if the <paramref name="instance"/> is an instance of an open type in the <paramref name="model"/>.
+        /// </summary>
+        /// <param name="instance">The instance to examine</param>
+        /// <param name="model">The client model.</param>
+        /// <returns>Returns true if the <paramref name="instance"/> is an instance of an open type in the <paramref name="model"/></returns>
+        internal static bool IsInstanceOfOpenType(object instance, ClientEdmModel model)
+        {
+            Debug.Assert(instance != null, "instance !=null");
+            Debug.Assert(model != null, "model !=null");
+
+            Type clientType = instance.GetType();
+            ClientTypeAnnotation clientTypeAnnotation = model.GetClientTypeAnnotation(clientType);
+
+            Debug.Assert(clientTypeAnnotation != null, "clientTypeAnnotation != null");
+
+            // TODO: Populate dynamic properties only if this is an open (entity or complex) type
+            return clientTypeAnnotation.EdmType.IsOpen();

```

Comment:
Correct me if I'm wrong. I thought we use //TODO to track tasks in visual studio that is pending.

Reply:
@paulodero I might have forgotten to purge some TODOs I had included during development. This is one of them and I'll remove it. I had added it there before implementing that check